code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -144,6 +145,31 @@
 	..()
 	icon_state = (ammo_magazine)? "arifle" : "arifle-empty"
 
+/obj/item/gun/projectile/automatic/rifle/carbine
+	name = "ballistic carbine"
+	desc = "A durable, rugged looking semi-automatic weapon of a make popular on the frontier worlds. Uses 5.56mm rounds and does not accept large \
+	capacity magazines. It is unmarked."
+	desc_fluff = "The ST24 is often considered the little brother of its larger and fully automatic counterpart, the STS35. It is a \
+	reliable and cheap to buy semi automatic carbine with many licensed manufacturers across the galaxy. It comes in different versions \
+	and calibres, this one uses 5.56 rounds. The manufacturer markings have been filed off."
+	icon = 'icons/obj/guns/bcarbine.dmi'
+	icon_state = "bcarbine"
+	item_state = "bcarbine"
+	caliber = "a556"
+	can_bayonet = TRUE
+	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 1, TECH_ILLEGAL = 3)
+	magazine_type = /obj/item/ammo_magazine/a556/carbine
+	allowed_magazines = list(/obj/item/ammo_magazine/a556/carbine)
+	knife_x_offset = 23
+	knife_y_offset = 13
+
+	firemodes = list(
+		list(mode_name="semiauto",	burst=1, fire_delay=10),)
```

Comment:
make this one line and removing the trailing comma

Reply:
got it king